stream-parent/stream-client/src/main/java/org/interledger/stream/sender/SimpleStreamSender.java

```diff
@@ -91,95 +117,170 @@ public SimpleStreamSender(
    */
   public SimpleStreamSender(
       final StreamEncryptionService streamEncryptionService, final Link link, ExecutorService executorService
+  ) {
+    this(streamEncryptionService, link, executorService, new StreamConnectionManager());
+  }
+
+  /**
+   * Required-args Constructor.
+   *
+   * @param streamEncryptionService A {@link StreamEncryptionService} used to encrypt and decrypted end-to-end STREAM
+   *                                packet data (i.e., packets that should only be visible between sender and
+   *                                receiver).
+   * @param link                    A {@link Link} that is used to send ILPv4 packets to an immediate peer.
+   * @param executorService         A {@link ExecutorService} to run the payments.
+   * @param streamConnectionManager A {@link StreamConnectionManager} that manages connections for all senders and
+   *                                receivers in this JVM.
+   */
+  public SimpleStreamSender(
+      final StreamEncryptionService streamEncryptionService,
+      final Link link,
+      final ExecutorService executorService,
+      final StreamConnectionManager streamConnectionManager
   ) {
     this.streamEncryptionService = Objects.requireNonNull(streamEncryptionService);
     this.link = Objects.requireNonNull(link);
 
     // Note that pools with similar properties but different details (for example, timeout parameters) may be
     // created using {@link ThreadPoolExecutor} constructors.
     this.executorService = Objects.requireNonNull(executorService);
+    this.streamConnectionManager = Objects.requireNonNull(streamConnectionManager);
   }
 
   @Override
   public CompletableFuture<SendMoneyResult> sendMoney(
-      final byte[] sharedSecret,
+      final SharedSecret sharedSecret,
       final InterledgerAddress sourceAddress,
       final InterledgerAddress destinationAddress,
       final UnsignedLong amount
   ) {
     Objects.requireNonNull(destinationAddress);
     Objects.requireNonNull(amount);
 
-    final SendMoneyAggregator sendMoneyAggregator = new SendMoneyAggregator(
+    final StreamConnection streamConnection = this.streamConnectionManager.openConnection(
```

Comment:
Per my comment above, `streamConnection` is never closed.

Reply:
hmmm...I think it is closed down on [line 363](https://github.com/hyperledger/quilt/pull/305/files#diff-2ec7bd39fe6f2a1ff349260aeddbdccaR363), no?

Reply:
Oh, my bad - the actual object. nm.

Reply:
For `v1.0`, @nhartner and I decided that the `SimpleStreamSender` will not close the Connection nor the Stream since the only receiver is a stateless receiver. We'll revisit this as part of #291